src/modules/languages/c.nix

```diff
@@ -6,16 +6,29 @@ in
 {
   options.languages.c = {
     enable = lib.mkEnableOption "tools for C development";
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.stdenv.cc;
+      example = "pkgs.clangStdenv.cc";
+      defaultText = "pkgs.stdenv.cc";
+      description = "C/C++ toolchain to use. Defaults to nixpkgs default C compiler (i.e. gcc on Linux and clang on macOS)";
+    };
+    languageServer = lib.mkOption {
+      type = lib.types.nullOr lib.types.package;
+      default = pkgs.clang-tools;
+      example = pkgs.ccls;
+      defaultText = "pkgs.clang-tools";
+      description = "Language server to use. Defaults to clangd from the clang-tools package.";
+    };
   };
 
   config = lib.mkIf cfg.enable {
     packages = with pkgs; [
-      stdenv
```

Comment:
I don't see what this package adds. It only contains nix-support hooks which I do not see being used here.

Reply:
I don't remember why I added it, should have added a comment. Let's see if it breaks something.